src/Symfony/Bridge/Doctrine/ConnectionTimingWeakMap.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine;
+
+class ConnectionTimingWeakMap
+{
+    private ?\WeakMap $connectionWeakMap = null;
+
+    public function __construct()
+    {
+        $this->connectionWeakMap = $this->connectionWeakMap ?? new \WeakMap();
+    }
+
+    public function setTimingPerConnection($connection, $connectionName): void
+    {
+        $timestamp = time();
+        $this->connectionWeakMap[$connection] = ['timing' => $timestamp, 'name' => $connectionName];
+    }
+
+    public function getTimingsPerConnection(): ?\WeakMap
```

Comment:
Using a WeakMap should be an implementation detail and not be given to the outside. It would be better to offer methods like `remove` and `all` etc. This also prevents someone from doing anything wrong with the WeakMap.

Reply:
Not sure about this: a WeakMap makes it clear that the design is made to break circular references. A class with a dedicated API doesn't make sense to me because this is only about wiring stuff, not providing an API.